src/Refinement/RefinementTransferOperators.jl

```diff
@@ -0,0 +1,166 @@
+
+"""
+  RefinementTransferOperator
+
+  Matrix-like operator providing a transfer between two `FESpaces`
+  defined in child-parent refined grids. 
+
+  The dofs are transfered by iteratively solving a projection problem 
+
+        (uh_to,vh_to)_to = (uh_from,vh_to)_Ω
+
+  where Ω is always chosen as the finest grid (i.e the RHS is
+  always integrated without loss of accuracy). 
+"""
+struct RefinementTransferOperator{T,A,B,C} <: AbstractMatrix{T}
+  from   ::A
+  to     ::B
+  caches ::C
+
+  function RefinementTransferOperator(T,from,to,caches)
```

Comment:
We may consider generalizing this data type so that one can leverage any `Gridap` `LinearSolver`, and not solely `cg!` from `IterativeSolvers`. Indeed, this would eliminate the added dependency among `Gridap` and `IterativeSolvers` here: https://github.com/gridap/Gridap.jl/pull/838/files#diff-72ed386c2a0cd1d23c0968297e70023ed98c22490d146dd89fc91f48369bad4dR16 (we actually would move that dependency to the tests)

Reply:
I agree, but how? I can add a `solver` field easily enough. However we would need this `solver` to have a common interface so that is can be called from within `mul!`. Something like `solve!(solver,x,y)` or something like that, right? I suppose we could use the `LinearSolver` type from Gridap, but I'm not sure if that's the way to go (it is designed to work on FEAffineOps, right?)

Reply:
>  (it is designed to work on FEAffineOps, right?)

Not necessarily. Look, e.g., at the implementation of the  `BackSlashSolver` or `LUSolver` subtyped of `LinearSolver`.

Reply:
Resolved by [4f93af9](https://github.com/gridap/Gridap.jl/pull/838/commits/4f93af96944d043d56677dd534829bc4e9a34e3a)